File: Day 003/202. Happy Number.java
class Solution {
    public boolean isHappy(int n) {
        int sum = n;
        while(sum != 1 && sum !=4){
            int t = sum;
            sum = 0;
            while(t!=0){
                int d=t%10;
                sum += d*d;
                t/= 10;
            }
        }
        return sum ==1;
    }
}

File: Day 003/202. Happy Number.py
class Solution(object):
    def isHappy(self, n):
        sum = n
        while sum != 1 and sum !=4 and sum !=9:
            t = sum
            sum =0
            while t !=0:
                d = t%10
                sum = sum + d*d
                t = t/10
        return sum == 1